profiles/apparmor/template.go

```diff
@@ -32,8 +32,6 @@ profile {{.Name}} flags=(attach_disconnected,mediate_deleted) {
   deny @{PROC}/sysrq-trigger rwklx,
   deny @{PROC}/kcore rwklx,
 
-  deny mount,
```

Comment:
should be removed only when CAP_SYS_ADMIN is granted?

Reply:
Wouldn't that leave `mount` as a special system call in the sense that it has a gate implemented for both mechanisms?  
Doesn't this lead to inconsistency and leave the option for other, similar errors?

Reply:
We already unblock syscalls in seccomp when caps are added. I think it makes sense to do the same here.

Reply:
Should this control in the apparmor template include the other syscalls gated by CAP_SYS_ADMIN in seccomp as well or shall we just add a check for unblocking `mount`?

Reply:
agree with Akihiro and Brian, these mechanisms are supposed to be independent. We should make them behave in the same way with capabilities.

Reply:
I understand, that sounds reasonable.

Reply:
I was thinking of adding the condition to the template.  
I'm still new to moby's source, but from what I see at  
https://github.com/moby/moby/blob/master/daemon/oci_linux.go#L954  
setting an apparmor profile and capabilities happens independently.
Is there a way to access the added capabilities from  
https://github.com/moby/moby/blob/master/profiles/apparmor/apparmor.go

Reply:
I've tried to understand the code better in the meanwhile.  
Isn't the default seccomp profile generated during container instantiation while the apparmor profile is created when the daemon starts? If so, this would block allowing `mount` in apparmor during container creation.  
Creating a new apparmor profile during container start based on the allowed capabilities seems to be an option, but as far as I see it doesn't fit the current design.

Reply:
I'm not particularly familiar with the code in this package.
I do seem to recall we used to generate these profiles on the fly at least to support privileged, I don't know what we are doing now.